### ISSUE TITLE
[BUG] Creating Docker Image Fails

### ISSUE THREAD

**Issue by tobiasduerschmid** [rosin-project/rosinstall_generator_time_machine#18]
When I run the build script for the docker file in the corresponding folder, I get the following error message (both when trying it on Windows and Ubuntu):

```
Sending build context to Docker daemon  5.632kB

Step 1/19 : FROM ubuntu:xenial
 ---> 065cf14a189c
Step 2/19 : ARG BUILD_VERSION
 ---> Using cache
 ---> 32895cf8eabb
Step 3/19 : LABEL maintainer="g.a.vanderhoorn@tudelft.nl"
 ---> Using cache
 ---> cc97ca251a13
Step 4/19 : LABEL org.label-schema.schema-version="1.0"
 ---> Using cache
 ---> 7148f6c6512b
Step 5/19 : LABEL org.label-schema.version=$BUILD_VERSION
 ---> Using cache
 ---> ccddc7ccc498
Step 6/19 : LABEL org.label-schema.vcs-url="https://github.com/rosin-project/rosinstall_generator_time_machine.git"
 ---> Using cache
 ---> c36a97a115a0
Step 7/19 : LABEL org.label-schema.name="robust-rosin/rosinstall_generator_time_machine"
 ---> Using cache
 ---> baaab84c5462
Step 8/19 : LABEL org.label-schema.description="Runtime infrastructure for rosinstall_generator_time_machine of the robust-rosin project."
 ---> Using cache
 ---> 30449bd745d7
Step 9/19 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> a83ae82ccb47
Step 10/19 : ENV LANG=C.UTF-8
 ---> Using cache
 ---> a583a7c0b812
Step 11/19 : ENV LC_ALL=C.UTF-8
 ---> Using cache
 ---> 62d1d01f799a
Step 12/19 : RUN apt-get update  && apt-get install -y --no-install-recommends       ca-certificates       git-core       python-distribute       python-pip       python-requests       python-setuptools       python-virtualenv       python-wheel       virtualenv  && apt-get clean  && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> edf8ac0f7ad9
Step 13/19 : WORKDIR /rgtm
 ---> Using cache
 ---> 3112ac8ddab3
Step 14/19 : RUN virtualenv -p python2.7 pre141  && virtualenv -p python2.7 post141
 ---> Using cache
 ---> 3224574e88b5
Step 15/19 : RUN /rgtm/pre141/bin/pip install -U pip  && /rgtm/pre141/bin/pip install -U wheel setuptools==20.1.1  && /rgtm/pre141/bin/pip install       git+https://github.com/rosin-project/rosdistro_python@rosin_bughunt_0.2.20#egg=rosdistro       rosinstall==0.7.2       rosdep==0.10.24       rosinstall_generator==0.1.5       catkin_pkg==0.4.16       rospkg==1.2.3       distribute==0.7.3       pyyaml==5.3       vcstools==0.1.30  && rm -rf /root/.cache/pip
 ---> Running in 30e4828dde09
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Requirement already up-to-date: pip in ./pre141/lib/python2.7/site-packages (20.3.4)
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Requirement already up-to-date: wheel in ./pre141/lib/python2.7/site-packages (0.36.2)
Collecting setuptools==20.1.1
  Downloading setuptools-20.1.1-py2.py3-none-any.whl (472 kB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 44.1.1
    Uninstalling setuptools-44.1.1:
      Successfully uninstalled setuptools-44.1.1
Successfully installed setuptools-20.1.1
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Collecting rosdistro
  Cloning https://github.com/rosin-project/rosdistro_python (to revision rosin_bughunt_0.2.20) to /tmp/pip-install-h9jZcf/rosdistro
  Running command git clone -q https://github.com/rosin-project/rosdistro_python /tmp/pip-install-h9jZcf/rosdistro
  Running command git checkout -b rosin_bughunt_0.2.20 --track origin/rosin_bughunt_0.2.20
  Switched to a new branch 'rosin_bughunt_0.2.20'
  Branch rosin_bughunt_0.2.20 set up to track remote branch rosin_bughunt_0.2.20 from origin.
Collecting rosinstall==0.7.2
  Downloading rosinstall-0.7.2.tar.gz (24 kB)
Collecting rosdep==0.10.24
  Downloading rosdep-0.10.24.tar.gz (67 kB)
    ERROR: Command errored out with exit status 1:
     command: /rgtm/pre141/bin/python2.7 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-h9jZcf/rosdep/setup.py'"'"'; __file__='"'"'/tmp/pip-install-h9jZcf/rosdep/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-VRyE0g
         cwd: /tmp/pip-install-h9jZcf/rosdep/
    Complete output (25 lines):
    failed to load symbols, rosdep will not function properly
    Cannot import rospkg, rosdep will not function properly
    Couldn't find index page for 'nose' (maybe misspelled?)
    No local packages or download links found for nose>=1.0
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-h9jZcf/rosdep/setup.py", line 29, in <module>
        license="BSD"
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/rgtm/pre141/local/lib/python2.7/site-packages/setuptools/dist.py", line 269, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/rgtm/pre141/local/lib/python2.7/site-packages/setuptools/dist.py", line 313, in fetch_build_eggs
        replace_conflicting=True,
      File "/rgtm/pre141/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 825, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/rgtm/pre141/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1070, in best_match
        return self.obtain(req, installer)
      File "/rgtm/pre141/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1082, in obtain
        return installer(requirement)
      File "/rgtm/pre141/local/lib/python2.7/site-packages/setuptools/dist.py", line 380, in fetch_build_egg
        return cmd.easy_install(req)
      File "/rgtm/pre141/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 634, in easy_install
        raise DistutilsError(msg)
    **distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('nose>=1.0')**
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by gavanderhoorn** [comment]
The ANSI escape codes made it a bit difficult to read. I've removed them.

This looks like it's another case of dependencies not having been pinned, like in #14. @ChrisTimperley fixes that by pinning things to known-working versions.

Could you try to perhaps install `nose` with a sufficiently old version *before* `rosdep` is installed?

If it's already present in the `venv`, it should not try to install it again with an unsupported version.

**Comment by gavanderhoorn** [comment]
This should be fixed by #20.
